## Tarea 2
### Materia: Laboratorio de Aprendizaje Estádistico 
### Profesor: Antonio Manguart Paez
### Alumna: Maria Fernanda Gómez Flores

Más informacióm del dataset 

El objetivo es predecir cuantas bicicletas se van a rentar

https://archive.ics.uci.edu/dataset/275/bike+sharing+dataset


Quita las variables de casual, registered, instant, fecha, mes, año de tu entrenamiento

Hay varias categoricas, cuales son? coviertelas e one hot encoding 

Crea variables dummy 

In [1]:
import pandas as pd
from sklearn import linear_model
from sklearn.metrics import r2_score 

In [2]:
df = pd.read_csv('../Data/day.csv')

In [3]:
y = df['cnt']

X = df.copy()
del X['cnt']
del X['casual']
del X['registered']
del X['instant']
del X['dteday']
del X['mnth']
del X['yr']

In [4]:
categoricas = [
    'season',
    'holiday',
    'weekday',
    'workingday',
    'weathersit'
]

In [5]:
X = pd.get_dummies(X, columns=categoricas)

In [6]:
model = linear_model.LinearRegression().fit(X,y)

In [7]:
predicciones_todo = model.predict(X)

In [8]:
r2_score(y_pred=predicciones_todo, y_true=y)

0.5642465831561199

El modelo obtuvo un \(R^2 = 0.5642\), por lo que explica aproximadamente el 56.42% de la variación en la cantidad de bicicletas rentadas.